File: Selenium/Activities/Activity21.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.chrome.service import Service
from webdriver_manager.chrome import ChromeDriverManager
from selenium.webdriver.support import expected_conditions as EC

service = Service(ChromeDriverManager().install())
driver = webdriver.Chrome(service=service)
wait = WebDriverWait(driver, 10)

driver.get("https://v1.training-support.net/selenium/tab-opener")
print("Home page title:", driver.title)

wait.until(EC.number_of_windows_to_be(2))
print("Currently open windows:", driver.window_handles)
for handle in driver.window_handles:
    driver.switch_to.window(handle)

wait.until(EC.element_to_be_clickable((By.ID, "actionButton")))
print("Current tab:", driver.current_window_handle)
print("Page title:", driver.title)
page_heading = driver.find_element(By.CLASS_NAME, "content").text
print("Page Heading:", page_heading)
driver.find_element(By.ID, "actionButton").click()

wait.until(EC.number_of_windows_to_be(3))
for handle in driver.window_handles:
    driver.switch_to.window(handle)

wait.until(EC.visibility_of_element_located((By.CLASS_NAME, "content")))
print("Current tab:", driver.current_window_handle)
print("Page title:", driver.title)
page_heading = driver.find_element(By.CLASS_NAME, "content").text
print("Page Heading:", page_heading)

driver.quit()

File: Selenium/Activities/Activity21.java
package seleniumActivities;

import java.time.Duration;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.By;
import org.openqa.selenium.support.ui.ExpectedConditions;

import io.github.bonigarcia.wdm.WebDriverManager;

public class Activity21 {

    public static void main(String[] args) {

        WebDriverManager.chromedriver().setup();
        WebDriver driver = new ChromeDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        
        driver.get("https://v1.training-support.net/selenium/tab-opener");
        System.out.println("Home page title: " + driver.getTitle());
        
        wait.until(ExpectedConditions.numberOfWindowsToBe(2));
        System.out.println("Currently open windows: " + driver.getWindowHandles());
        for(String handle : driver.getWindowHandles()) {
            driver.switchTo().window(handle);
        }

        wait.until(ExpectedConditions.elementToBeClickable(By.id("actionButton")));
        System.out.println("Current tab: " + driver.getWindowHandle());
        System.out.println("Page title: " + driver.getTitle());
        String pageHeading = driver.findElement(By.className("content")).getText();
        System.out.println("Page Heading: " + pageHeading);
        driver.findElement(By.id("actionButton")).click();

        wait.until(ExpectedConditions.numberOfWindowsToBe(3));
        for(String handle : driver.getWindowHandles()) {
            driver.switchTo().window(handle);
        }

        wait.until(ExpectedConditions.visibilityOfElementLocated(By.className("content")));
        System.out.println("Current tab: " + driver.getWindowHandle());
        System.out.println("Page title: " + driver.getTitle());
        pageHeading = driver.findElement(By.className("content")).getText();
        System.out.println("Page Heading: " + pageHeading);

        driver.quit();
    }
}
